### PR TITLE
Replace PropertyGraph in cugraph-PyG with FeatureStore

### LINKED ISSUE #3079: [FEA] cuGraph-PyG Should Pass Canonical Edge Type to New Uniform Neighbor Sample Function
Currently, cuGraph-PyG only passes the cuGraph edge type, but this adds unnecessary overhead.  Instead, the canonical edge type should be passed as an integer.  A lookup table may need to be added in `CuGraphStore` to handle this.

Partially resolved by #3101

### LINKED ISSUE #3151: [IMP] Migrate all PropertyGraph code in cugraph-PyG to FeatureStore
cuGraph-PyG is no longer relying on PropertyGraph for functionality.  All reliance on PropertyGraph should be replaced with calls to FeatureStore.

Due: 1/18/2023 6:00 PM EST

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
Replaces PropertyGraph in cugraph-PyG with the new FeatureStore, which is the preferred solution for storing tensor features.
Closes #3151 
Closes #3079

### CODE REVIEW

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:354]: Does this mean that you always have to create a pandas dataframe for using PyG. I think we should allow a path to provide cudf and dask_cudf dataframes directly as that will probably be much more scalable.

File: python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_sampler.py
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_sampler.py:190]:
```suggestion

```

File: python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py:150]: I would suggest using the to_dlpack object here or using `cp.asarray(a)` . 

https://pytorch.org/docs/stable/dlpack.html#torch.utils.dlpack.to_dlpack

```suggestion
        index = cudf.from_dlpack(index.__dlpack__())
```

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:606]: Same suggestion here.

File: python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_store.py
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_store.py:236]: Same suggestion here.

**Review by VibhuJawa — CHANGES_REQUESTED** [review]

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:354]: I have a separate issue to allow passing in a cudf or dask_cudf dataframe.  That won't be part of this PR.

Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:354]: See #3176

File: python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py:150]: This doesn't work for cudf

Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py]: resolved, all dlpack calls removed

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: resolved, all dlpack calls removed

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:52]: Motivation of this fix . Should have been clearer, this gives you managed memory for free with Pytorch as dlpack  fails .   

```python
import rmm      
import torch
import cudf
rmm.reinitialize(managed_memory= True)
torch.cuda.memory.change_current_allocator(rmm.rmm_torch_allocator)
```
Below works:
```python
ar = cudf.Series([1]).repeat(1_000_000_000).values
t1 = torch.as_tensor(ar, device='cuda')
t1
tensor([1, 1, 1,  ..., 1, 1, 1], device='cuda:0')
```
Below Fails: 
```python
ar = cudf.Series([1]).repeat(1_000_000_000).values
t2 = torch.from_dlpack(ar.toDlpack())
```
```python
RuntimeError: Unsupported device_type: 13
```

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Minor review on using `cuda-array-interface`

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:52]: I thought I changed this

Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: @VibhuJawa the push failed, please review again.

**Review by VibhuJawa — APPROVED** [review]
LGTM

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py
Comment (VibhuJawa) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: Thanks. Approved it .

File: python/cugraph-pyg/cugraph_pyg/data/__init__.py
Comment (rlratzel) [inline python/cugraph-pyg/cugraph_pyg/data/__init__.py:1]: I didn't catch this in PR 3170: should this year range start at 2022?

**Review by rlratzel — APPROVED** [review]
Noticed one minor item which need not hold up approval.

**Pull request merged**
